src/stream/src/executor/managed_state/join/mod.rs

```diff
@@ -445,10 +464,12 @@ impl<K: HashKey, S: StateStore> JoinHashMap<K, S> {
                 let degree_i64 = degree_row
                     .datum_at(degree_row.len() - 1)
                     .expect("degree should not be NULL");
-                entry_state.insert(
-                    pk,
-                    JoinRow::new(row.into_owned_row(), degree_i64.into_int64() as u64).encode(),
-                );
+                entry_state
+                    .insert(
+                        pk,
+                        JoinRow::new(row.into_owned_row(), degree_i64.into_int64() as u64).encode(),
+                    )
+                    .expect("duplicated pk");
```

Comment:
?

Reply:
What about `row.to_owned_row()`?

Reply:
Didn't really get this comment... In the original code, the row is already passed to `entry_state.insert`, right? So what do you mean by "extra copy"?

Reply:
<img width="1765" alt="image" src="https://github.com/risingwavelabs/risingwave/assets/31772373/3f9fce61-427d-4b71-859d-6ba6320bf4ff">
If we change it to `to_owned_row`
<img width="1765" alt="image" src="https://github.com/risingwavelabs/risingwave/assets/31772373/3099b79e-eaed-4683-9f10-59580c96be40">

Reply:
Oops. Why converting to `OwnedRow` here? `JoinRow` takes generic `R`.

https://github.com/risingwavelabs/risingwave/blob/376846c758cc1455b32bd8255b9a6ef2cc0eaaa5/src/stream/src/executor/managed_state/join/mod.rs#L60-L63